test/pkgcheck.sh

```diff
@@ -40,24 +112,55 @@ cd btmp1
   Darwin)
     export LDFLAGS="-Wl,-headerpad_max_install_names"
     ;;
-  Linux)
-    if grep -i fedora /etc/os-release > /dev/null
-    then
-        # Note: Fedora patches cmake to use -O2 in release, which
-        # does not match the -O3 configure sets :-(
-        export CFLAGS="-O2 -DNDEBUG"
-    fi
-    ;;
   esac
-  ../configure
+  # Use same optimization level as cmake did.
+  CFLAGS="$(awk -F= '/CMAKE_C_FLAGS_RELEASE:|CMAKE_C_FLAGS:/ {print $2}' < ../btmp2/CMakeCache.txt | tr '\012' ' ')"
+  export CFLAGS
+  ../configure $CONFIGURE_ARGS
   make
   make install
 cd ..
 
-if diff --exclude '*.so*' --exclude '*.a' -Nur pkgtmp1 pkgtmp2
+repack_ar() {
+  if ! cmp --silent pkgtmp1/usr/local/lib/libz$suffix.a pkgtmp2/usr/local/lib/libz$suffix.a
+  then
+    echo "libz$suffix.a does not match.  Probably filenames differ (.o vs .c.o).  Unpacking and renaming..."
+    # Note: %% is posix shell syntax meaning "Remove Largest Suffix Pattern", see
+    # https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_06_02
+    cd pkgtmp1; ar x usr/local/lib/libz$suffix.a; rm usr/local/lib/libz$suffix.a; cd ..
+    cd pkgtmp2; ar x usr/local/lib/libz$suffix.a; rm usr/local/lib/libz$suffix.a; for a in *.c.o; do mv $a ${a%%.c.o}.o; done; cd ..
+  fi
+}
+
+case $(uname) in
+Darwin)
+  # Alas, dylibs still have an embedded hash or something,
+  # so nuke it.
+  # FIXME: find a less fragile way to deal with this.
```

Comment:
Is there something we can do about this? How many bytes were changed?

Reply:
Per the dd command, 16 bytes.

One does not do anything about Apple tools, one endures them :-)

Reply:
Would be nice to have some link to site that describes what those bytes actually are... Mach-O file format reference doesn't have clear definition what starts at byte offset 1337 as the file header isn't that long...

Reply:
Yes, that would be nice.  However, I don't know how to dig it up myself, and I don't know that it's worth holding up this pull request for.

Reply:
I'm not sure if this is current but `strip` seems to have a `-no_uuid` flag that will strip out all `LC_UUID` load commands.
https://www.manpagez.com/man/1/strip/

Reply:
That sounds handy.  Also, evidently ld has a -no_uuid option.  I'll see if those can replace the dd!

Reply:
So far no joy.  It was (and maybe is) a good theory, though.

I just refreshed the mac support a bit, turned up an interesting bit so I left it as a separate commit.

Reply:
Wait, there is joy in mudville.  Stand by...

Reply:
I got it to go green, but there's a catch.  See https://github.com/zlib-ng/zlib-ng/issues/681